### ISSUE TITLE
Different results with same data

### ISSUE THREAD

**Issue by UnibsMatt** [nebuly-ai/nebullvm#39]
Hi everyone. I'm having some troubles getting consistency throughout the same inputs tensor..
Hw Sw overview:
- Ubuntu 18.04
- TensorRT-8.4.0.6
- llvm 14.0
- pip installed torch, onnx tensorflow latest stable
- cuda 11.6
- cudnn 8.4.0
- nvidia 3070tx
- Intel(R) Core(TM) i7-9700E CPU @ 2.60GHz

The optimization that i'm doing as the samples:
```

import torch
import torchvision.models as models
from nebullvm import optimize_torch_model
import time
import numpy as np

model = models.vgg19()
bs, input_sizes = 1, [(3, 224, 224)]
save_dir = "."
optimized_model = optimize_torch_model( model, batch_size=bs, input_sizes=input_sizes, save_dir=save_dir)

x = torch.rand((1, *input_sizes[0]))

with torch.no_grad():
    res = optimized_model(x)[0]
```

every iteration with the same vector give different results.. Sometimes there are NaN values and sometimes they change based on odd or even number of iterations. 
Example res[0][:10]:

```
tensor([1.4013e-45, 4.5877e-41, 1.4013e-45,        nan, 1.4013e-45, 7.0065e-45, 1.4013e-45, 1.4013e-44, 2.9427e-44, 1.4013e-44])
tensor([3.6121e-11, 4.5877e-41, 3.6121e-11, 4.5877e-41, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00])
tensor([ 1.4013e-45,  0.0000e+00, -1.8365e+00,  0.0000e+00,  3.3631e-44, 0.0000e+00,  3.0810e-17,  4.5877e-41,  1.4013e-45,  4.5877e-41])
tensor([ 1.4013e-45,  0.0000e+00, -1.8370e+00,  0.0000e+00,  1.4013e-45, 0.0000e+00, -1.8370e+00,  0.0000e+00,  3.3631e-44,  0.0000e+00])

```

Same x vector different results, sometimes about 40 order of magnitude. I'm getting something wrong?
The inference time for the vgg19 it's incredible but I'm scared the something under the hood it's not working properly :

```
%%timeit
res = optimized_model(x)[0]
145 µs ± 48 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

```

**Comment by morgoth95** [comment]
Thanks @UnibsMatt for pointing out the error. Could you please share with us the whole logs you are getting during the optimization? Thanks in advance!

BTW, if you have any other problem you can easily get in touch with us in our [discord channel](https://discord.gg/jyjtZTPyHS)!

**Comment by morgoth95** [comment]
Hello @UnibsMatt! We found a bug in our TensorRT code. We fixed it in release 0.2.1! You can install it and try again! 
PS. Remember to set the PyTorch model in evaluation mode before optimising it: just add `model = model.eval()` to your code before running the `optimize_torch_model` function!

**Comment by kartikeyporwal** [comment]
> Hi everyone. I'm having some troubles getting consistency throughout the same inputs tensor.. Hw Sw overview:
> 
> * Ubuntu 18.04
> * TensorRT-8.4.0.6
> * llvm 14.0
> * pip installed torch, onnx tensorflow latest stable
> * cuda 11.6
> * cudnn 8.4.0
> * nvidia 3070tx
> * Intel(R) Core(TM) i7-9700E CPU @ 2.60GHz
> 
> The optimization that i'm doing as the samples:
> 
> ```
> 
> import torch
> import torchvision.models as models
> from nebullvm import optimize_torch_model
> import time
> import numpy as np
> 
> model = models.vgg19()
> bs, input_sizes = 1, [(3, 224, 224)]
> save_dir = "."
> optimized_model = optimize_torch_model( model, batch_size=bs, input_sizes=input_sizes, save_dir=save_dir)
> 
> x = torch.rand((1, *input_sizes[0]))
> 
> with torch.no_grad():
>     res = optimized_model(x)[0]
> ```
> 
> every iteration with the same vector give different results.. Sometimes there are NaN values and sometimes they change based on odd or even number of iterations. Example res[0][:10]:
> 
> ```
> tensor([1.4013e-45, 4.5877e-41, 1.4013e-45,        nan, 1.4013e-45, 7.0065e-45, 1.4013e-45, 1.4013e-44, 2.9427e-44, 1.4013e-44])
> tensor([3.6121e-11, 4.5877e-41, 3.6121e-11, 4.5877e-41, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00, 0.0000e+00])
> tensor([ 1.4013e-45,  0.0000e+00, -1.8365e+00,  0.0000e+00,  3.3631e-44, 0.0000e+00,  3.0810e-17,  4.5877e-41,  1.4013e-45,  4.5877e-41])
> tensor([ 1.4013e-45,  0.0000e+00, -1.8370e+00,  0.0000e+00,  1.4013e-45, 0.0000e+00, -1.8370e+00,  0.0000e+00,  3.3631e-44,  0.0000e+00])
> ```
> 
> Same x vector different results, sometimes about 40 order of magnitude. I'm getting something wrong? The inference time for the vgg19 it's incredible but I'm scared the something under the hood it's not working properly :
> 
> ```
> %%timeit
> res = optimized_model(x)[0]
> 145 µs ± 48 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
> ```

I think this could apply to PyTorch model also when we don't run `model.eval()`

Even `np.allclose(
    model(x)[0].detach().numpy(),
    model(x)[0].detach().numpy()
)` will lead to a `False` when don't eval the model.

**Comment by UnibsMatt** [comment]
Thanks very much I'll try this morning and let you know. Thanks for the response
